### PR TITLE
Lw 10449 sdk e2e tests ci

### PR DESCRIPTION
**Pull request by ce0la**
# Context
Background
An easy to run CI action should be available to quickly sanitize SRE deployments (e.g. dev-preprod).

DoD
CI action can be run targeting different deployments.

An env variable (or such) should be configurable to determine which deployment we are targeting

…
https://input-output.atlassian.net/browse/LW-10449
_Reason for the change? If an issue exists, reference it here using a [keyword](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)_

# Proposed Solution

# Important Changes Introduced

### CODE REVIEW

File: .github/workflows/continuous-integration-e2e.yaml [hunk 1]
```diff
@@ -42,6 +42,18 @@ jobs:
...
+        inputs:
+          network:
+            description: "Network to update (dev-mainnet, dev-preprod, etc.)"
+            required: true
+            type: string
+
+      - name: Generate .env file
+        working-directory: ./packages/e2e/
+        run: |
+          KEY_MANAGEMENT_PARAMS=${{ secrets.KEY_MANAGEMENT_PARAMS }}
```
Comment (gytis-ivaskevicius) [inline .github/workflows/continuous-integration-e2e.yaml:54]: this one changes depending on network

File: .github/workflows/continuous-integration-e2e.yaml [hunk 2]
```diff
@@ -50,6 +62,7 @@ jobs:
 
       - name: 🔨 Build
         run: |
+          source .env
```
Comment (gytis-ivaskevicius) [inline .github/workflows/continuous-integration-e2e.yaml:65]: why do we need to source .env everywhere?

File: packages/e2e/src/scripts/generate-network-dotenv.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+network="$1"
```
Comment (gytis-ivaskevicius) [inline packages/e2e/src/scripts/generate-network-dotenv.sh:3]: technically this is an `environment` or `release_train`

**Review by gytis-ivaskevicius — CHANGES_REQUESTED** [review]
KEY_MANAGEMENT_PARAMS needs to be fixed, other than that if it works - its good

File: .github/workflows/test-deploy-e2e.yaml [hunk 4]
```diff
@@ -1,15 +1,73 @@
...
+        env:
+          CARDANO_NODE_CHAINDB_LOG_LEVEL: 'Warning'
+          CARDANO_NODE_LOG_LEVEL: 'Warning'
+          OGMIOS_PORT: '1340'
+          OGMIOS_URL: 'ws://ogmios:1340'
+          POSTGRES_PORT: '5435'
+
+      - name: Wait for network init
+        run: |
+          yarn workspace @cardano-sdk/e2e wait-for-network-init
```
Comment (iccicci) [inline .github/workflows/test-deploy-e2e.yaml:69]: Sorry, I forgot: my bad.

These two steps must be removed.

File: .github/workflows/test-deploy-e2e.yaml [hunk 5]
```diff
@@ -1,15 +1,73 @@
-name: CI/CD
+name: Continuous Integration - SDK E2E
```
Comment (iccicci) [inline .github/workflows/test-deploy-e2e.yaml:1]: I know I'm being nitpicky 😉 but this is not a `Continuous Integration` action, what about `Deployed environments - SDK E2E tests`?

**Review by iccicci — COMMENTED** [review]
Some notes... but great job! Thank you

File: .github/workflows/test-deploy-e2e.yaml [hunk 6]
```diff
@@ -1,15 +1,73 @@
...
     steps:
-      - name: Hello World
+      - name: 📥 Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Generate .env file
+        working-directory: ./packages/e2e/
+        run: |
+          networkMagic=$(jq --arg environment "${{ inputs.environment }}" <<< '{"dev-preprod":1, "dev-preview":2, "dev-mainnet":764824073, "staging-preprod":1}' '.[$environment]')
+          ./src/scripts/generate-dotenv.sh ${{ inputs.environment }}
```
[the remark this answers is not in the document]
Comment (iccicci) [reply .github/workflows/test-deploy-e2e.yaml:40]: `dev-mainnet` is useless ATM, some environment in input is missing.

[the remark this answers is not in the document]
Comment (ce0la) [reply .github/workflows/test-deploy-e2e.yaml:40]: A simple conditional would suffice for the two targeted networks (preprod and preview). Pushing the change now.

[the remark this answers is not in the document]
Comment (ce0la) [reply .github/workflows/test-deploy-e2e.yaml]: https://github.com/input-output-hk/cardano-js-sdk/pull/1332/commits/9b9b6d36da52b133bee7707f3a581a8ca0e41452

**Review by gytis-ivaskevicius — DISMISSED** [review]

**Review by iccicci — APPROVED** [review]
Exceptional job! Thank you!

**Review by gytis-ivaskevicius — APPROVED** [review]
Good job :)

**Review by michalrus — APPROVED** [review]
Nice! Thank you :)

**Pull request merged**